pkg/gateway/gateway.go

```diff
@@ -299,7 +410,31 @@ func (g *gatewayModule) configPath(name string) string {
 	return filepath.Join(g.proxyConfigPath, fmt.Sprintf("%s.yaml", name))
 }
 
-func (g *gatewayModule) SetNamedProxy(wlID string, prefix string, backends []string, TLSPassthrough bool) (string, error) {
+func (g *gatewayModule) validateNameContract(name string, twinID uint32) error {
+	contractID, err := g.sub.GetContractIDByNameRegistration(string(name))
+	if errors.Is(err, substrate.ErrNotFound) {
+		return errors.New("a name contract must be reserved first")
+	}
+	if err != nil {
+		return err
+	}
+	contract, err := g.sub.GetContract(contractID)
+	if errors.Is(err, substrate.ErrNotFound) {
```

Comment:
While you return 3 different errors but each one should be handled differently. for example, a NotFound and `!IsCreated` should decomission the domain, but any other error should be "logged" but don't act up on because it can be just a network glitch